pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AccessorMethodGenerationRule.java

```diff
@@ -66,4 +100,33 @@ public void analyzeMember(final AbstractJavaAccessNode node, final List<NameOccu
             }
         }
     }
+
+    public boolean isCompileTimeConstant(ASTPrimaryExpression expressions) {
+        // function call detected
+        List<ASTPrimarySuffix> suffix = expressions.findDescendantsOfType(ASTPrimarySuffix.class);
+        if (!suffix.isEmpty()) {
+            return false;
+        }
+
+        // single node expression
+        List<ASTName> nameNodes = expressions.findDescendantsOfType(ASTName.class);
+        List<ASTLiteral> literalNodes = expressions.findDescendantsOfType(ASTLiteral.class);
+        if (nameNodes.size() + literalNodes.size() < 2) {
+            for (ASTName node: nameNodes) {
+                if (cache.containsKey(node.getImage())) {
+                    return cache.get(node.getImage());
+                }
+            }
+            return true;
```

Comment:
this is dangerous… given the example:

```java
public class Outer {
    private static final int FOO = Inner.BAR;

    public static class Inner {
        private static final int BAR = new Random().nextInt();
    }
} 
```

As `Inner.BAR` hasn't been resolved yet, it will default to saying it's a constant, which it's not.

Reply:
Any sample on how to use symbol table will be really helpful.

Reply:
All symboltable info is available through the scopes objects. This very class makes extensive usage of that (see line 37 for obtaining the global file scope and how it is navigated down). The current code doesn't make it very easy to look for stuff such as `Inner.BAR` (would require to first resolve `Inner` manually in the current scope, and then looking for `BAR` there), but we are planning on revamping it for 7.0.0.

As I said, I'd leave that for a separate PR for the sake of keeping this small.